runtime/compiler/env/j9method.cpp

```diff
@@ -7020,9 +7045,23 @@ TR_ResolvedJ9Method::getResolvedDynamicMethod(TR::Compilation * comp, I_32 callS
          targetJ9MethodBlock = fej9()->targetMethodFromMemberName((uintptr_t) fej9()->getReferenceElement(*invokeCacheArray, JSR292_invokeCacheArrayMemberNameIndex)); // this will not work in AOT or JITServer
          // if the callSite table entry is resolved, we can check if the appendix object is null,
          // in which case the appendix object must not be pushed to stack
-         uintptr_t appendixObject = (uintptr_t) fej9()->getReferenceElement(*invokeCacheArray, JSR292_invokeCacheArrayAppendixIndex);
+         appendixObject = (uintptr_t) fej9()->getReferenceElement(*invokeCacheArray, JSR292_invokeCacheArrayAppendixIndex);
          if (isInvokeCacheAppendixNull && !appendixObject) *isInvokeCacheAppendixNull = true;
          }
+
+      if (comp->compileRelocatableCode())
+         {
+         bool valid =
+            comp->getSymbolValidationManager()->addDynamicMethodFromCallsiteIndex(
+               targetJ9MethodBlock,
+               getNonPersistentIdentifier(),
+               callSiteIndex,
+               !appendixObject);
+
+         if (!valid)
+            comp->failCompilation<J9::AOTHasInvokeHandle>("Failed to add valiation rcord for resolved dynamic method %p", targetJ9MethodBlock);
```

Comment:
Failing compilation seems a bit drastic. Would it be hard to treat it as unresolved? Similarly in `getResolvedHandleMethod()`

I do see that we already fail compilation when `!isInvokeCacheEntryAnArray()`, but I think we should probably treat it as unresolved if we can in that case too (not that that should be part of this PR)

Reply:
See I was thinking about that, but it's not as straightforward to revert to unresolved. The reason is because I would need `isUnresolvedCallSiteTableEntry`/`isUnresolvedMethodTypeTableEntry` to then also return `true` for this cpIndex.

It's definitely something to consider looking into but I'd rather do it in another PR since it's not a trivial change to make.

Reply:
OK, yeah, I thought it might not be so straightforward